pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -196,6 +211,18 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 		c.stopController(statusKey, statusStopChan)
 	}
 
+	typeConfig.Status.ObservedGeneration = typeConfig.Generation
+
+	typeConfig.Status.PropagationController = corev1a1.ControllerStatusRunning
```

Comment:
These two fields are both set to running - which may not be the correct value. You should set these fields based on the actual state of these controllers.

Reply:
Yeah, I made changes in the code to accommodate this requirement.